src/test/components/credentialsForm/credentialsForm.test.js

```diff
@@ -116,71 +95,49 @@ describe('CredentialsForm', () => {
+        <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
+        store,
+        initialEntry
+      )
+    );
 
-    expect(wrapper.find(Modal).props().title).toEqual('View account authorization credentials');
-    expect(wrapper.find(Modal).props().description).toEqual(undefined);
-    expect(wrapper.find(Modal).props().footer).toEqual(null);
+    await waitFor(() => expect(screen.getByText('View account authorization credentials')).toBeInTheDocument());
 
-    expect(
-      wrapper
-        .find(SourcesFormRenderer)
-        .props()
-        .schema.fields.every((field) => field.isDisabled)
-    ).toEqual(true);
+    expect([...screen.getAllByRole('textbox')].every((e) => e.disabled)).toEqual(true);
   });
 
   it('closes via cross icon', async () => {
-    await act(async () => {
-      wrapper = mount(
-        componentWrapperIntl(
-          <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
-          store,
-          initialEntry
-        )
-      );
-    });
-    wrapper.update();
+    render(
+      componentWrapperIntl(
+        <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
+        store,
+        initialEntry
+      )
+    );
 
-    await act(async () => {
-      wrapper.find(Button).first().simulate('click');
-    });
-    wrapper.update();
+    await waitFor(() => expect(() => screen.getByRole('progressbar')).toThrow());
 
-    expect(wrapper.find(MemoryRouter).instance().history.location.pathname).toEqual(
-      replaceRouteId(routes.sourcesDetail.path, sourceId)
-    );
+    userEvent.click(screen.getByLabelText('Close'));
+
+    expect(screen.getByTestId('location-display').textContent).toEqual(replaceRouteId(routes.sourcesDetail.path, sourceId));
   });
 
   it('closes via cancel button', async () => {
-    await act(async () => {
-      wrapper = mount(
-        componentWrapperIntl(
-          <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
-          store,
-          initialEntry
-        )
-      );
-    });
-    wrapper.update();
+    render(
+      componentWrapperIntl(
+        <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
+        store,
+        initialEntry
+      )
+    );
 
-    await act(async () => {
-      wrapper.find(Button).last().simulate('click');
-    });
-    wrapper.update();
+    await waitFor(() => expect(() => screen.getByRole('progressbar')).toThrow());
 
-    expect(wrapper.find(MemoryRouter).instance().history.location.pathname).toEqual(
-      replaceRouteId(routes.sourcesDetail.path, sourceId)
-    );
+    userEvent.click(screen.getByText('Cancel'));
+
+    expect(screen.getByTestId('location-display').textContent).toEqual(replaceRouteId(routes.sourcesDetail.path, sourceId));
```

Comment:
This can be bypassed by calling
```
screen.getByRole('WHAT_IS_THE_ELEMENT?', { name: replaceRouteId(routes.sourcesDetail.path, sourceId) })
```

Reply:
The current url is rendered as a child of the element. Location-display is a helper to get the current location displayed.